django/core/management/commands/migrate.py

```diff
@@ -67,6 +67,11 @@ def add_arguments(self, parser):
             '--check', action='store_true', dest='check_unapplied',
             help='Exits with a non-zero status if unapplied migrations exist.',
         )
+        parser.add_argument(
+            '--prune', action='store_true', dest='prune',
+            help='Removes migrations from the django_migrations table that '
+                 'are no longer present on disk.',
+        )
```

Comment:
I think that's fair. I went back and forth on this, because this is mostly intended to be used with squashed migrations, and we do encourage re-running `migrate` in order to finish recording squashed migrations as fully applied, but I'll have another look and see if the recorder still runs without having to apply any migrations 👍🏻

Reply:
I guess I won't worry about that for now, it's probably "least surprise" for it it to return early without migrating/recording.